src/xenia/vfs/devices/stfs_container_device.cc

```diff
@@ -301,16 +337,27 @@ StfsContainerDevice::Error StfsContainerDevice::ReadEntrySVOD(
   entry_address += true_ordinal_offset;
 
   // Read block's descriptor
-  auto data = mmap_.at(entry_file)->data() + entry_address;
 
-  uint16_t node_l = xe::load<uint16_t>(data + 0x00);
-  uint16_t node_r = xe::load<uint16_t>(data + 0x02);
-  uint32_t data_block = xe::load<uint32_t>(data + 0x04);
-  uint32_t length = xe::load<uint32_t>(data + 0x08);
-  uint8_t attributes = xe::load<uint8_t>(data + 0x0C);
-  uint8_t name_length = xe::load<uint8_t>(data + 0x0D);
-  auto name_buffer = reinterpret_cast<const char*>(data + 0x0E);
-  auto name = std::string(name_buffer, name_length);
+  auto file = files_.at(entry_file);
+  xe::filesystem::Seek(file, entry_address, SEEK_SET);
+
+  uint16_t node_l;
```

Comment:
Perhaps a locally defined struct to read into, as opposed to a batch set of variables?